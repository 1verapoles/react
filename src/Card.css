.card {
  max-width: 18%;
  margin-right: 20px;
  margin-bottom: 20px;
}