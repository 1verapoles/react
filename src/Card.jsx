import React from 'react'
import PropTypes from 'prop-types';
import { Link } from "react-router-dom";
import './Card.css'
import imgDef from './assets/images/comp.jpg'

function Card({ title, description, author, publishedAt, urlToImage, url }) {
  let date = new Date(publishedAt).toLocaleString().split(',')[0];
  let titleUrl = encodeURIComponent(title);
  return (
    <Link className="card" to={`/details/${titleUrl}`}>
      <img src={urlToImage || imgDef} className="card-img-top" alt={title} />
      <div className="card-body card-body2">
        <h5 className="card-title">{title}</h5>
        <p className="card-text">{description}</p>
      </div>
      <ul className="list-group list-group-flush">
        <li className="list-group-item">Автор: {author}</li>
        <li className="list-group-item">Дата публикации: {date}</li>
      </ul>
    </Link>
  )
}

Card.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  author: PropTypes.string,
  publishedAt: PropTypes.string,
  url: PropTypes.string,
  urlToImage: PropTypes.string
};

export default Card
