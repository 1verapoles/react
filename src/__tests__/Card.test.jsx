import React from 'react';
import { BrowserRouter } from "react-router-dom";
import { render } from '@testing-library/react';
import Card from '../Card';

describe('Testing Card component', () => {
  it('renders without crashing', () => {
    const { asFragment } = render(<BrowserRouter><Card /></BrowserRouter>);

    expect(asFragment()).toMatchSnapshot();
  });
});
