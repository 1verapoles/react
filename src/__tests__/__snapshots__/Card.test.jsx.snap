// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Card component renders without crashing 1`] = `
<DocumentFragment>
  <a
    class="card"
    href="/details/undefined"
  >
    <img
      class="card-img-top"
      src="[object Object]"
    />
    <div
      class="card-body card-body2"
    >
      <h5
        class="card-title"
      />
      <p
        class="card-text"
      />
    </div>
    <ul
      class="list-group list-group-flush"
    >
      <li
        class="list-group-item"
      >
        Автор: 
      </li>
      <li
        class="list-group-item"
      >
        Дата публикации: Invalid Date
      </li>
    </ul>
  </a>
</DocumentFragment>
`;
